### DISCUSSION TITLE
onChange function with useInput

### DISCUSSION THREAD

**Discussion by iantbarker** [gpbl/react-day-picker#1487] [Support] [answered]
I feel like I'm missing something, but I can't figure out how to pass an onChange type callback function using useInput so I can use the value in a parent component? I can't find anything in the docs.

https://codesandbox.io/s/react-day-picker-useinput-smutz7

**Comment by iantbarker** [comment]
I ended up using a useEffect with a `dayPickerProps.selected` dependency and that seems to be working for what I need it to do.

**Comment by gpbl** [comment, marked as answer]
@iantbarker not sure what you are trying to do... something like this? https://codesandbox.io/s/react-day-picker-useinput-forked-i9c5ue?file=/index.js

**Reply by iantbarker** [reply]
@gpbl  I am wanting to pass a callback to a component that is a wrapper. So selecting a date in the picker would fire the callback. Playing with your example, it looks like I can do 
`<DayPicker {...dayPickerProps} onDayClick={(e) => console.log(3, e)} />`

And achieve what I want... Does that seem right?

**Reply by gpbl** [reply]
Sure, you can use the same approach - just remember to call the `dayPickerProps.onDayClick` in your callback:

https://codesandbox.io/s/react-day-picker-useinput-forked-i9c5ue?file=/index.js:604-817

```tsx
<DayPicker
  {...dayPickerProps}
  onDayClick={(day, activeModifiers, e) => {
    console.log("onDayClick", day);
    dayPickerProps.onDayClick(day, activeModifiers, e);
  }}
/>
```

**Reply by iantbarker** [reply]
Ah. Super helpful. Thank you.
